src/rdma/BaseRDMA.cc

```diff
@@ -83,33 +83,48 @@ void BaseRDMA::createBuffer() {
     ifstream numa_node_file;
     numa_node_file.open(std::string(dev_list[i]->ibdev_path)+"/device/numa_node");
     int numa_node = -1;
-    numa_node_file >> numa_node;
+    if (numa_node_file) {
+      numa_node_file >> numa_node;
+    } else {
+      numa_node = 0;
```

Comment:
I don't think 0 is a good default value for this case. This was included by accident to try out RXE on my local machine where the numa_node file didn't exist and the numa_node for the parent net device was -1 (i.e. not assigned by the kernel driver for some reason)

Reply:
I've created #16 partly based on your changed here, since I have similar problems on ARM cores with the numa_node and device selection.
Maybe it works for you as well when selecting the device by name.

Reply:
Thanks! I'd need to try it out some more. There were multiple other issues down the line, but the basic layout in terms of `numa_node` was that there's a `/sys/class/infiniband/rxe_XXX` folder that had no `device` symlink, but a `parent` file that contains the name of a `/sys/class/net/<name>` interface, which in turn has a `device` symlink with a `numa_node` file that was originally `-1`, but can be written to; to set it to some value (e.g. `0`)